src/background/cookie-instrument.ts

```diff
@@ -21,52 +57,34 @@ export class CookieInstrument {
       /** The underlying reason behind the cookie's change. */
       cause: OnChangedCause;
     }) => {
-      const change = changeInfo.removed ? "deleted" : "added";
-
-      // TODO: Support other cookie operations
-      if (change === "deleted" || change === "added" || change === "changed") {
-        const update = {} as JavascriptCookieChange;
-
-        update.change = change;
-        update.change_cause = changeInfo.cause;
-        update.crawl_id = crawlID;
-
-        const cookie: Cookie = changeInfo.cookie;
-
-        // Expiry time (in seconds)
-        // A comment from pre-webextension code, which may still be valid:
-        // May return ~Max(int64). I believe this is a session
-        // cookie which doesn't expire. Sessions cookies with
-        // non-max expiry time expire after session or at expiry.
-        const expiryTime = cookie.expirationDate; // returns seconds
-        let expiryTimeString;
-        if (!cookie.expirationDate || expiryTime === 9223372036854776000) {
-          expiryTimeString = "9999-12-31T21:59:59.000Z";
-        } else {
-          const expiryTimeDate = new Date(expiryTime * 1000); // requires milliseconds
-          expiryTimeString = expiryTimeDate.toISOString();
-        }
-        update.expiry = expiryTimeString;
-        update.is_http_only = boolToInt(cookie.httpOnly);
-        update.is_host_only = boolToInt(cookie.hostOnly);
-        update.is_session = boolToInt(cookie.session);
-
-        update.host = escapeString(cookie.domain);
-        update.is_secure = boolToInt(cookie.secure);
-        update.name = escapeString(cookie.name);
-        update.path = escapeString(cookie.path);
-        update.value = escapeString(cookie.value);
-        update.same_site = escapeString(cookie.sameSite);
-        update.first_party_domain = escapeString(cookie.firstPartyDomain);
-
-        update.time_stamp = new Date().toISOString();
-
-        this.dataReceiver.saveRecord("javascript_cookies", update);
-      }
+      const eventType = changeInfo.removed ? "deleted" : "added-or-changed";
+      const update: JavascriptCookieRecord = {
+        record_type: eventType,
+        change_cause: changeInfo.cause,
+        crawl_id: crawlID,
+        extension_session_uuid: extensionSessionUuid,
+        ...transformCookieObjectToMatchOpenWPMSchema(changeInfo.cookie),
+      };
+      this.dataReceiver.saveRecord("javascript_cookies", update);
     };
     browser.cookies.onChanged.addListener(this.onChangedListener);
   }
 
+  public async saveAllCookies(crawlID) {
```

Comment:
Indeed, this was added in relation to https://github.com/motin/jestr-pioneer-shield-study/issues/2. Added it to the instrumentation rather than JESTr so that privacy-related extensions can perform something similar to the cookie table dumping that OpenWPM performs using the profile contents on disk.

Reply:
Cool. Added https://github.com/mozilla/openwpm-webext-instrumentation/issues/28 to remind us to document this at some point.